app/controllers/datasets_controller.rb

```diff
@@ -13,54 +23,75 @@ def create
     if @dataset.save
       redirect_to new_licence_dataset_path(@dataset)
     else
-      render 'new'
+      render 'dataset'
+    end
+  end
+
+  def update
+    @dataset = current_dataset
+    @dataset.update_attributes(params.require(:dataset).permit(:title, :summary, :description))
+
+    if @dataset.save
+      redirect_to dataset_path(@dataset)
+    else
+      render 'dataset'
     end
   end
 
   def licence
     @dataset = current_dataset
 
-    if request.post?
+    unless request.get?
       licence = get_licence(params.require(:dataset).permit(:licence, :licence_other))
       @dataset.licence = licence
 
-      redirect_to new_location_dataset_path(@dataset) if @dataset.save
+      if @dataset.save
+        redirect_to new_location_dataset_path(@dataset) if request.post?
```

Comment:
really what i should do is seperate these into four methods (new, edit, create, update)